rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,226 @@
+     * @return 1 SUCCESS, -1 ADDRESS_NOT_EXIST, -2 INVALID_ADDRESS_FORMAT, -10 UNAUTHORIZED_CALLER,
+     * 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int removeLockWhitelistAddress(Transaction tx, String addressBase58) {
+        if (isNotAuthorizedLockWhitelistChange(tx)) {
+            return UNAUTHORIZED_CALLER.getCode();
+        }
+
+        LockWhitelist whitelist = whitelistStorageProvider.getLockWhitelist();
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            if (!whitelist.remove(address)) {
+                return ADDRESS_NOT_EXIST.getCode();
+            }
+            return SUCCESS.getCode();
+        } catch (AddressFormatException e) {
+            logger.error("Invalid Address Format error in removeLockWhitelistAddress: {}", e.getMessage());
+            return INVALID_ADDRESS_FORMAT.getCode();
+        } catch (Exception e) {
+            logger.error("Unexpected error in removeLockWhitelistAddress: {}", e.getMessage());
+            return UNKNOWN_ERROR.getCode();
+        }
+    }
+
+    /**
+     * Sets a delay in the BTC best chain to disable lock whitelist
+     *
+     * @param tx                  current RSK transaction
+     * @param disableBlockDelayBI block since current BTC best chain height to disable lock
+     *                            whitelist
+     * @return 1 SUCCESS, -1 DELAY_ALREADY_SET, -2 DISABLE_BLOCK_DELAY_INVALID, -10 UNAUTHORIZED_CALLER
+     */
+    @Override
+    public int setLockWhitelistDisableBlockDelay(Transaction tx, BigInteger disableBlockDelayBI, int btcBlockchainBestChainHeight)
+        throws IOException, BlockStoreException {
+
+        if (isNotAuthorizedLockWhitelistChange(tx)) {
+            return UNAUTHORIZED_CALLER.getCode();
+        }
+
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist();
+        if (lockWhitelist.isDisableBlockSet()) {
+            return DELAY_ALREADY_SET.getCode();
+        }
+
+        int disableBlockDelay = disableBlockDelayBI.intValueExact();
+        if (disableBlockDelay + btcBlockchainBestChainHeight <= btcBlockchainBestChainHeight) {
+            return DISABLE_BLOCK_DELAY_INVALID.getCode();
+        }
+
+        lockWhitelist.setDisableBlockHeight(btcBlockchainBestChainHeight + disableBlockDelay);
+
+        return SUCCESS.getCode();
+    }
+
+    @Override
+    public boolean verifyLockSenderIsWhitelisted(Address senderBtcAddress, Coin totalAmount, int height) {
+        // If the address is not whitelisted, then return the funds
+        // using the exact same utxos sent to us.
+        // That is, build a pegout waiting for confirmations and get it in the pegoutWaitingForConfirmations set.
+        // Otherwise, transfer SBTC to the sender of the BTC
+        // The RSK account to update is the one that matches the pubkey "spent" on the first bitcoin tx input
```

Comment:
I think this comment might be out of place.
It seems this class knows too many details of the `pegout` process. It shouldn't. This class should only know how to whitelist, disable and enable addresses, etc.